src/worker/syscalls.rs

```diff
@@ -37,16 +38,45 @@ impl Worker for SyscallsWorker {
     fn run_payload(&self) -> Result<(), WorkerError> {
         info!("{self}");
 
-        let Workload::Syscalls { arrival_rate } = self.workload.workload else {
+        let mut counter = 0;
+        let mut start = Instant::now();
+
+        let Workload::Syscalls {
+            arrival_rate,
+            tight_loop,
+            syscall_nr,
+        } = self.workload.workload
+        else {
             unreachable!()
         };
 
+        let syscall = Sysno::from(syscall_nr);
+        info!("Running syscall {syscall}");
+
         loop {
             let worker = *self;
-            thread::spawn(move || {
-                worker.do_syscall().unwrap();
-            });
 
+            if start.elapsed().as_secs() > 10 {
+                warn!(
+                    "CPU {}, {}",
+                    self.config.cpu.id,
+                    counter / start.elapsed().as_secs()
+                );
+                start = Instant::now();
+                counter = 0;
+            }
+
+            counter += 1;
+            // Do the syscall directly, without spawning a thread (it would
+            // introduce too much overhead for a quick syscall).
+            worker.do_syscall(syscall).unwrap();
```

Comment:
Panicking here is probably fine, but returning an error could be a bit more user friendly.